### ISSUE TITLE
Cannot search for strings with spaces

### ISSUE THREAD

**Issue by wavexx** [firecat53/tabview#114]
Calling gather() at each character in search has the side-effect of stripping spaces while you're still writing them. The backspace handling is also broken, since it should be str[-1].strip().lower() to avoid a similar issue.

Looking at how textpad.gather() is implemented though, it just fetches the text in the current window's buffer so it has no way of knowing the difference. Frankly, curses.textpad is next to useless.... reminds me of the handling as done in bank's terminal forms from the 80'ties, where everything was fixed-size and god-forbid about moving the cursor!

**Comment by firecat53** [comment]
Yep, we're going to have to revert to getch to fix this and the Unicode
entry issue. I added a couple of links in the Unicode issue for other
projects' handling of text entry.

**Comment by wavexx** [comment]
Note that you could "fix" the current code by calling getyx before gather() and setyx() just after to restore the cursor position. I tried it last day, but it feels like punching a dead horse.
